mllib/src/test/scala/org/apache/spark/ml/util/MLTest.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.util
+
+import java.io.File
+
+import org.scalatest.Suite
+
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.Transformer
+import org.apache.spark.sql.{DataFrame, Encoder, Row}
+import org.apache.spark.sql.execution.streaming.MemoryStream
+import org.apache.spark.sql.streaming.StreamTest
+import org.apache.spark.sql.test.TestSparkSession
+import org.apache.spark.util.Utils
+
+trait MLTest extends StreamTest with TempDirectory { self: Suite =>
+
+  @transient var sc: SparkContext = _
+  @transient var checkpointDir: String = _
+
+  protected override def createSparkSession: TestSparkSession = {
+    new TestSparkSession(new SparkContext("local[2]", "MLlibUnitTest", sparkConf))
+  }
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sc = spark.sparkContext
+    checkpointDir = Utils.createDirectory(tempDir.getCanonicalPath, "checkpoints").toString
+    sc.setCheckpointDir(checkpointDir)
+  }
+
+  override def afterAll() {
```

Comment:
In https://github.com/apache/spark/pull/9677/files#diff-2d609a0839a51280e0f10cc73ef42d21R35 we added a call to clear the active SQLContext after each MLlib test suite, and MLlibTestSparkContext still does that: https://github.com/apache/spark/blob/8ae004b4602266d1f210e4c1564246d590412c06/mllib/src/test/scala/org/apache/spark/mllib/util/MLlibTestSparkContext.scala#L49

At the time, that was necessary to avoid flakiness in Jenkins tests.  Do you know if that's no longer necessary?  CCing @nkronenfeld and @gatorsmile  who seem to have worked on the SQL test traits recently

Reply:
Maybe not needed if you are not hitting any flaky test

Reply:
Well, we'll find out in a few weeks  : )

Reply:
Checking back in here, I haven't seen flakiness, but I have seen cascading failures, which I believe are a new phenomenon: A failure in one test suite A seems to cause subsequent tests suites B, C, ... to fail.  (But when A is fixed, then B, C,... run correctly.)  Do you not see this in SQL tests?  Do you think this might be related to not cleaning up the active context?

Reply:
Actually, it's worse than this.  I see a bunch of failures when I run multiple test suites at once, even when doing `sbt clean package` beforehand and without any tests which fail by themselves.  Will test on master and complain on the dev list if it's an issue.  (No need to respond here)